Sources/AppStoreConnect/Networking/Response.swift

```diff
@@ -133,38 +79,103 @@ public struct Response<Received: Equatable & Sendable>: Equatable, Sendable {
     }
 
     /// Checks the response for any unacceptable properties, such as a non-2XX status code, in order to verify integrity.
-    /// - Throws: A ``Response/Error`` if the status code is not indicative of success.
-    package func check() throws {
+    /// - Throws: A ``ResponseError`` if the status code is not indicative of success.
+    public func check() throws {
         switch statusCode {
         case 429:
-            throw Error.rateLimitExceeded(error: errorResponse, rate: rate, response: response)
+            throw ResponseError.rateLimitExceeded(error: errorResponse, rate: rate, response: response)
         case ..<200, 300...:
-            throw Error.requestFailure(error: errorResponse, statusCode: statusCode, response: response)
+            throw ResponseError.requestFailure(error: errorResponse, statusCode: statusCode, response: response)
         default:
             break
         }
     }
 
-    /// Decodes the ``data`` into the intended container type.
+    /// Decodes the `data` into the intended container type.
     /// - Parameter decoder: The decoder to decode the data with.
     /// - Returns: The decoded data.
-    /// - Throws: An error if ``data`` is `nil` or cannot be decoded as the expected type.
-    func decode<Output>(using decoder: JSONDecoder) throws -> Output where Output: Decodable, Received == Data {
+    /// - Throws: An error if `data` is `nil` or cannot be decoded as the expected type.
+    public func decode<Output>(using decoder: JSONDecoder) throws -> Output where Output: Decodable, Received == Data {
         guard let data = data else {
-            throw Error.dataAssertionFailed
+            throw ResponseError.dataAssertionFailed
         }
 
         return try decoder.decode(Output.self, from: data)
     }
 
-    /// Unwraps the stored ``URL`` and returns it.
+    /// Unwraps the stored `URL` and returns it.
     /// - Returns: URL to a downloaded file on-disk.
-    /// - Throws: An error if ``data`` is nil.
-    func decode() throws -> URL where Received == URL {
+    /// - Throws: An error if `data` is nil.
+    public func decode() throws -> URL where Received == URL {
         guard let data = data else {
-            throw Error.dataAssertionFailed
+            throw ResponseError.dataAssertionFailed
         }
 
         return data
     }
 }
+
+/// Error thrown when the response from the ``Transport`` fails to meet expectations.
+public enum ResponseError: Swift.Error, Equatable {
+    /// The response as requested was unsuccessful, as described by the `statusCode` and `error`.
+    case requestFailure(error: ErrorResponse?, statusCode: StatusCode, response: URLResponse)
+    /// The client has exceeded the rate limit for the current period.
+    case rateLimitExceeded(error: ErrorResponse?, rate: Rate?, response: URLResponse)
+    /// Data was expected to exist or conform to some format but failed.
+    case dataAssertionFailed
+
+    var isRecoverable: Bool {
```

Comment:
Moving that to the strategy? :)  And the logic for retry on the `Rate` type (public), and let folks do what they want in their Strategy. ?